File: app/src/main/java/be/vergauwen/simon/cleanmvp/ui/TestComponent.java
package be.vergauwen.simon.cleanmvp.ui;

import be.vergauwen.simon.cleanmvp.di.ActivityScope;
import be.vergauwen.simon.cleanmvp.di.ApplicationComponent;
import be.vergauwen.simon.cleanmvp.mvp.MVPContract;
import dagger.Component;

@ActivityScope
@Component(dependencies = ApplicationComponent.class)
public interface TestComponent extends MVPContract.Component<TestPresenter> {
    TestPresenter presenter();
}

File: app/src/main/kotlin/be/vergauwen/simon/ui/TestComponent.kt
package be.vergauwen.simon.ui

import be.vergauwen.simon.di.ActivityScope
import be.vergauwen.simon.di.ApplicationComponent
import be.vergauwen.simon.mvp.MVPContract
import dagger.Component

@ActivityScope
@Component(dependencies = arrayOf(ApplicationComponent::class))
interface TestComponent : MVPContract.Component<TestContract.View, TestPresenter> {
  override fun presenter(): TestPresenter
}
